index.js

```diff
@@ -15,12 +16,42 @@ const proxyHandler = {
   }
 }
 
-function proxyFreeze (target) {
+const identifiableProxyHandler = {
+  ...proxyHandler,
+  get: (target, key) => {
+    if (key === proxyIdentifier) return true
+    return target[key]
+  }
+}
+
+function proxyFreeze (target, { addProxyIdentifier = false, preventRefreeze = false } = {}) {
   if (typeof target !== 'function' && typeof target !== 'object') {
     throw new Error('proxyFreeze only supports constructor functions or objects.')
   }
+  if (typeof addProxyIdentifier !== 'boolean') {
+    throw new TypeError('addProxyIdentifier needs to be a boolean.')
+  }
+  if (typeof preventRefreeze !== 'boolean') {
+    throw new TypeError('preventRefreeze needs to be a boolean.')
+  }
+  if (preventRefreeze && !addProxyIdentifier) {
+    throw new Error('cannot use preventRefreeze without addProxyIdentifier.')
+  }
+
+  if (target && target[proxyIdentifier]) {
+    if (preventRefreeze) return target
+
+    const msg = 'Trying to freeze an already frozen object.'
+    if (typeof process === 'undefined') {
+      (console.warn.bind(console) || console.error.bind(console) || console.log.bind(console))(msg)
+    } else {
+      process.emitWarning(msg, 'ProxyFreezeWarning')
+    }
```

Comment:
nit: factor this out to `emitWarning` fn

Reply:
45fd7bb16c7fea4db12095e89a20509d5177bcfc